generateBins.py

```diff
@@ -28,59 +18,28 @@
 PFPR_FILE       = "rwa_pfpr2to10.asc"
 POPULATION_FILE = "rwa_population.asc"
 
-
 # TODO Determine the bins computationally
 # Following bins are for Rwanda
 POPULATION_BINS = [2125, 5640, 8989, 12108, 15577, 20289, 27629, 49378, 95262, 286928]
 
-# getting data ready for binning
-# data should be 1-dimensional array, python list or iterable
-myArray  = np.loadtxt("GIS\\rwa_population.asc", skiprows = 6)
-print (type(myArray))
-print(myArray.ndim)
-print(myArray.shape)
-print(myArray.size)
-array_1d = myArray.flatten()
-print (array_1d)
-print(array_1d.ndim)
-print(array_1d.shape)
-print(array_1d.size)
-#gvf = 0.0
-#nclasses = 20
-
-def goodness_of_variance_fit(array, classes):
-    # get the break points
-    classes = jenkspy.jenks_breaks(array, classes)
-
-    # do the actual classification
-    classified = np.array([classify(i, classes) for i in array])
-
-    # max value of zones
-    maxz = max(classified)
-
-    # nested list of zone indices
-    zone_indices = [[idx for idx, val in enumerate(classified) if zone + 1 == val] for zone in range(maxz)]
-
-    # sum of squared deviations from array mean
-    sdam = np.sum((array - array.mean()) ** 2)
-
-    # sorted polygon stats
-    array_sort = [np.array([array[index] for index in zone]) for zone in zone_indices]
-
-    # sum of squared deviations of class means
-    sdcm = sum([np.sum((classified - classified.mean()) ** 2) for classified in array_sort])
-
-    # goodness of variance fit
-    gvf = (sdam - sdcm) / sdam
-
-    return gvf
-
-
 def process(configuration, gisPath = ""):
     # Load the configuration
     cfg = load_configuration(configuration)
+    filename = os.path.join(gisPath, PFPR_FILE)
+
+    # getting data ready for binning
+    # data should be 1-dimensional array, python list or iterable
+    myArray = np.loadtxt(load_asc(filename), skiprows=6)
+    array_1d = myArray.flatten()
 
     # TODO Add the stuff for the population bins!
+    # GVF Implementation
+
+    gvf = goodness_of_variance_fit(array_1d, 14)
+    print("The value of Goodness of Variance fit is:", gvf)
+    if (gvf < 0.7):
+        print("Warning: GVF too low")
```

Comment:
Alright, on the right track, but think of something more along the lines of `find_best_bins(data, minimum, maximum)` that returns the best number of bins from that range and the GVF. Then you can point the GVF if need be and call another scripts like `get_breaks(data, number_of_bins)` to get the breaks that are used in place of `POPULATION_BINS`.